lib/cylc/config.py

```diff
@@ -782,7 +782,9 @@ def _expand_runtime(self):
                     new_environ = OrderedDictWithDefaults()
                     for p_name, p_val in indices.items():
                         p_var_name = 'CYLC_TASK_PARAM_%s' % p_name
-                        new_environ[p_var_name] = p_val
+                        new_param = OrderedDictWithDefaults()
+                        new_param[p_var_name] = p_val
+                        self.task_param[name] = new_param
```

Comment:
I noticed the branch doesn't work properly with multiple parameters, e.g. `foo<m,n>`.
  * new_param doesn't need to be ordered dict, a plain dict will do `{}`
  * new_param should be defined outside the loop, we need to store `task_param_vars[task-name] = {p_var_name_1: p_val_1, p_var_name_2:p_val_2, ...}